src/js/sw/sw.js

```diff
@@ -217,3 +220,53 @@ const fetchHandler = async (event) => {
 self.addEventListener('install', precacheAssets);
 self.addEventListener('activate', clearOldCaches);
 self.addEventListener('fetch', fetchHandler);
+
+/** @type {MessagePort} */
+let messageChannelPort;
+
+self.addEventListener(
+  'message',
+  (event) => {
+    if (event.data.type === 'channel-port') {
+      [messageChannelPort] = event.ports;
+    }
+  },
+);
+
+const bgFetchHandler = async (e) => {
+  /** @type {BackgroundFetchRegistration} */
+  const bgFetchRegistration = e.registration;
+  const bgFetch = new BackgroundFetch();
```

Comment:
Nit: Shall we move this until we actually need `videoId`?

```js

const bgFetchHandler = async (e) => {
  /** @type {BackgroundFetchRegistration} */
  const bgFetchRegistration = e.registration;
  const records = await bgFetchRegistration.matchAll();
  const urls = records.map((record) => record.request.url);
  
  if (urls.length === 0)
    return;
  
  const responsePromises = records.map((record) => record.responseReady);
  const responses = await Promise.all(responsePromises);
    
  const bgFetch = new BackgroundFetch();
  bgFetch.fromRegistration(bgFetchRegistration);
...
```

Reply:
@beaufortfrancois Good suggestion, thanks! Addressed in 7979e32.